corehq/apps/enterprise/enterprise.py

```diff
@@ -383,3 +388,68 @@ def _get_individual_export_rows(self, exports, export_line_counts):
             )
 
         return rows
+
+
+class EnterpriseCommCareVersionReport(EnterpriseReport):
+    title = _('CommCare Client Version Compliance')
+    metric = _('%% of mobile workers on the latest commcare client version ')
+
+    def __init__(self, account, couch_user, **kwargs):
+        super().__init__(account, couch_user, **kwargs)
+
+    @property
+    def headers(self):
+        return [
+            _('Mobile Worker'),
+            _('Project Space'),
+            _('Latest Version Available at Submission'),
+            _('Version in Use'),
+
+        ]
+
+    def rows_for_domain(self, domain_obj):
+        rows = []
+
+        for user in get_all_user_rows(domain_obj.name,
+                                    include_web_users=False,
+                                    include_mobile_users=True,
+                                    include_inactive=False,
+                                    include_docs=True):
+            user = CouchUser.wrap_correctly(user['doc'])
+            last_submission = user.reporting_metadata.last_submission_for_user
+            last_used_device = user.last_device
+
+            version_in_use = _('Unknown')
+            date_of_use = None
+
+            # If the user hasn't submitted a form, we use the last used device
+            if last_submission and last_submission.commcare_version:
+                version_in_use = _format_commcare_version(last_submission.commcare_version)
+                date_of_use = last_submission.submission_date
+            elif last_used_device and last_used_device.commcare_version:
+                version_in_use = _format_commcare_version(last_used_device['commcare_version'])
+                date_of_use = last_used_device.last_used
+
+            latest_version_at_time_of_use = get_latest_version_at_time(date_of_use)
+
+            if is_out_of_date(version_in_use, latest_version_at_time_of_use):
+                rows.append([
+                    user.username,
+                    domain_obj.name,
+                    latest_version_at_time_of_use,
+                    version_in_use,
+                ])
+
+        return rows
+
+    @property
+    def total(self):
+        total_mobile_workers = 0
+        total_up_to_date = 0
+        for domain_obj in self.domains():
+            domain_mobile_workers = get_mobile_user_count(domain_obj.name, include_inactive=False)
+            if domain_mobile_workers:
+                total_mobile_workers += domain_mobile_workers
+                total_up_to_date += domain_mobile_workers - len(self.rows_for_domain(domain_obj))
+        percentage = (total_up_to_date / total_mobile_workers) if total_mobile_workers else 0
+        return f"{percentage * 100:.0f}%"
```

Comment:
Change to 1 decimal place

Reply:
c9586dc2534329070e385a95e7bd1914a8b379a7
I also make it a function to format the percentage, so both you and I can reuse it, thus making sure the percentage will always have just one digit.

Reply:
Nice, thanks!